engine/chain/internal/arc/broadcast.go

```diff
@@ -0,0 +1,106 @@
+package arc
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net/http"
+
+	sdk "github.com/bitcoin-sv/go-sdk/transaction"
+	"github.com/bitcoin-sv/spv-wallet/engine/chain/errors"
+	"github.com/bitcoin-sv/spv-wallet/engine/chain/internal/ef"
+	"github.com/bitcoin-sv/spv-wallet/engine/chain/models"
+	"github.com/bitcoin-sv/spv-wallet/engine/spverrors"
+	"github.com/go-resty/resty/v2"
+)
+
+// Custom ARC defined http status codes
+const (
+	StatusNotExtendedFormat             = 460
+	StatusFeeTooLow                     = 465
+	StatusCumulativeFeeValidationFailed = 473
+)
+
+// Broadcast submits a transaction to the ARC server and returns the transaction info.
+func (s *Service) Broadcast(ctx context.Context, tx *sdk.Transaction) (*chainmodels.TXInfo, error) {
+	result := &chainmodels.TXInfo{}
+	arcErr := &chainmodels.ArcError{}
+	req := s.prepareARCRequest(ctx).
+		SetResult(result).
+		SetError(arcErr)
+
+	s.setCallbackHeaders(req)
+
+	txHex, err := s.prepareTxHex(ctx, tx)
+	if err != nil {
+		return nil, err
+	}
+
+	req.SetBody(requestBody{
+		RawTx: txHex,
+	})
+
+	response, err := req.Post(fmt.Sprintf("%s/v1/tx", s.arcCfg.URL))
+
+	if err != nil {
+		return nil, s.wrapRequestError(err)
+	}
+
+	switch response.StatusCode() {
+	case http.StatusOK:
+		if result.TXStatus.IsProblematic() {
+			return nil, chainerrors.ErrARCProblematicStatus.Wrap(spverrors.Newf("ARC Problematic tx status: %s", result.TXStatus))
+		}
+		return result, nil
+	case http.StatusUnauthorized, http.StatusForbidden, http.StatusNotFound:
+		return nil, s.wrapARCError(chainerrors.ErrARCUnauthorized, arcErr)
+	case http.StatusConflict:
+		return nil, s.wrapARCError(chainerrors.ErrARCGenericError, arcErr)
```

Comment:
You can remove this, I believe Arc is never returning that :)

Reply:
Removed.